lib/dust.js

```diff
@@ -248,17 +248,27 @@
   };
 
   /**
-   * Decide somewhat-naively if something is a Thenable.
+   * Decide somewhat-naively if something is a Thenable.  Matches Promises A+ Spec, section 1.2 “thenable” is an object or function that defines a then method."
    * @param elem {*} object to inspect
    * @return {Boolean} is `elem` a Thenable?
    */
   dust.isThenable = function(elem) {
-    return elem &&
-           typeof elem === 'object' &&
+    return elem &&  /* Beware: `typeof null` is `object` */
```

Comment:
I had a think about this:  When `elem` is `null` it'll be falsy, so it'll early-out on the `elem && ...`.  Testing for `null` would be redundant (albeit explicit).

There look to be a few tests that ensure this e.g. "scalar null in a # section", so it looks covered.